core/backend/src/test/IModelTestUtils.ts

```diff
@@ -575,17 +588,1160 @@ export class IModelTestUtils {
+    // delete PhysicalObject3
+    const physicalObjectId3: Id64String = queryByUserLabel(sourceDb, "PhysicalObject3");
+    assert.isTrue(Id64.isValidId64(physicalObjectId3));
+    sourceDb.elements.deleteElement(physicalObjectId3);
+    assert.equal(Id64.invalid, queryByUserLabel(sourceDb, "PhysicalObject3"));
+    // Insert PhysicalObject5
+    const physicalObjectProps5: PhysicalElementProps = {
+      classFullName: PhysicalObject.classFullName,
+      model: physicalElement1.model,
+      category: spatialCategoryId,
+      code: Code.createEmpty(),
+      userLabel: "PhysicalObject5",
+      geom: createBox(Point3d.create(1, 1, 1)),
+      placement: {
+        origin: Point3d.create(5, 5, 5),
+        angles: YawPitchRollAngles.createDegrees(0, 0, 0),
+      },
+    };
+    const physicalObjectId5: Id64String = sourceDb.elements.insertElement(physicalObjectProps5);
+    assert.isTrue(Id64.isValidId64(physicalObjectId5));
+    // delete relationship
+    const drawingGraphicId1: Id64String = queryByUserLabel(sourceDb, "DrawingGraphic1");
+    const drawingGraphicId2: Id64String = queryByUserLabel(sourceDb, "DrawingGraphic2");
+    const relationship: Relationship = sourceDb.relationships.getInstance(DrawingGraphicRepresentsElement.classFullName, { sourceId: drawingGraphicId2, targetId: physicalObjectId1 });
+    relationship.delete();
+    // insert relationships
+    DrawingGraphicRepresentsElement.insert(sourceDb, drawingGraphicId1, physicalObjectId5);
+    DrawingGraphicRepresentsElement.insert(sourceDb, drawingGraphicId2, physicalObjectId5);
+    // update InformationRecord2
+    const informationRecordCodeSpec: CodeSpec = sourceDb.codeSpecs.getByName("InformationRecords");
+    const informationModelId = sourceDb.elements.queryElementIdByCode(InformationPartitionElement.createCode(sourceDb, subjectId, "Information"))!;
+    const informationRecodeCode2: Code = new Code({ spec: informationRecordCodeSpec.id, scope: informationModelId, value: "InformationRecord2" });
+    const informationRecordId2: Id64String = sourceDb.elements.queryElementIdByCode(informationRecodeCode2)!;
+    assert.isTrue(Id64.isValidId64(informationRecordId2));
+    const informationRecord2: any = sourceDb.elements.getElement(informationRecordId2);
+    informationRecord2.commonString = `${informationRecord2.commonString}-Updated`;
+    informationRecord2.sourceString = `${informationRecord2.sourceString}-Updated`;
+    informationRecord2.update();
+    // delete InformationRecord3
+    const informationRecodeCode3: Code = new Code({ spec: informationRecordCodeSpec.id, scope: informationModelId, value: "InformationRecord3" });
+    const informationRecordId3: Id64String = sourceDb.elements.queryElementIdByCode(informationRecodeCode3)!;
+    assert.isTrue(Id64.isValidId64(informationRecordId3));
+    sourceDb.elements.deleteElement(informationRecordId3);
+  }
+
+  export async function prepareTargetDb(targetDb: IModelDb): Promise<void> {

```

Comment:
These methods don't really seem general enough to be in here... What was the reasoning behind moving them out of the TransformationUtils into this file?

Reply:
yeah, with a name like `prepareTargetDb`, it is somewhat transformer specific. There are some plain backend tests (in `DeleteDefinitionElements.test.ts` that use `prepareSourceDb` even not in a transformation related test, which motivated me to move everything that didn't directly depend on the transformation package. I don't think I should move those tests out of the backend package, maybe renaming them to `prepareIModelScenario1` is better? and then possibly change the `prepareTargetDb` to `prepareIModelScenario2` and put that in the transformer package's personal test utilities?

Reply:
do you have any alternatives?

Reply:
If it is going to stay here, ideally something more specifc to what kind of iModel it is creating. I haven't reviewed all the code for creation but I assume it's building up an iModel that stresses some specific piece. Knowing what that piece is would be helpful for anyone going to use it.

Reply:
I chose `ExtensiveTestScenario`, it is a non-exhaustive amalgam of various elements and schema items for testing different schema and element processing cases, for instance, the transform tests use it to test converting elements between classes, among other things. While originally for testing certain transformer cases, it is used in the DeleteDefinitionElements test and I would consider it more of a non-exhaustive collection of interesting cases. I can't think of a more specific name myself, unless we split the model into several smaller models. I also think it's valuable having a feature-exhaustive iModel to run tests on, to hit more edge cases, so I vote we don't split it.